### PR TITLE
python3Packages.pythran: 0.9.8post3 -> 0.9.12

### PR DESCRIPTION
**Pull request by FRidh**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/pythran/default.nix [hunk 1]
```diff
@@ -12,19 +13,31 @@
...
     repo = "pythran";
     rev = version;
-    sha256 = "sha256-GCWjJlf7zpFzELR6wTF8FoJzJ3F/WdT1hHjY5A5h/+4=";
+    sha256 = "sha256-lQbVq4K/Q8RzlFhE+l3HPCmUGmauXawcKe31kfbUHsI=";
   };
 
+  patches = [
+    (substituteAll {
+      src = ./0001-hardcode-path-to-libgomp.patch;
+      gomp = "${stdenv.cc.cc.lib}/lib/libgomp${stdenv.hostPlatform.extensions.sharedLibrary}";
```
Comment (FRidh) [inline pkgs/development/python-modules/pythran/default.nix:36]: maybe not correct for cross?

File: pkgs/development/python-modules/pythran/default.nix [hunk 2]
```diff
@@ -12,19 +13,30 @@
...
     owner = "serge-sans-paille";
     repo = "pythran";
     rev = version;
-    sha256 = "sha256-GCWjJlf7zpFzELR6wTF8FoJzJ3F/WdT1hHjY5A5h/+4=";
+    sha256 = "sha256-lQbVq4K/Q8RzlFhE+l3HPCmUGmauXawcKe31kfbUHsI=";
   };
 
+  patches = [
+    (substituteAll {
+      src = ./0001-hardcode-path-to-libgomp.patch;
```
Comment (FRidh) [inline pkgs/development/python-modules/pythran/default.nix:35]: patch will break non-gcc omp

Comment (FRidh) [reply pkgs/development/python-modules/pythran/default.nix:35]: maybe add a `passthru.openmp` to all compilers?

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 130464` at 4dc94726 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages marked as broken and skipped:</summary>
  <ul>
    <li>python38Packages.tensorflowWithCuda</li>
    <li>python39Packages.tensorflowWithCuda</li>
  </ul>
</details>
<details>
  <summary>25 packages skipped due to time constraints:</summary>
  <ul>
    <li>python38Packages.arviz</li>
    <li>python38Packages.baselines</li>
    <li>python38Packages.dm-sonnet</li>
    <li>python38Packages.edward</li>
    <li>python38Packages.graph_nets</li>
    <li>python38Packages.mask-rcnn</li>
    <li>python38Packages.optuna</li>
    <li>python38Packages.pymc3</li>
    <li>python38Packages.rl-coach</li>
    <li>python38Packages.scikit-tda</li>
    <li>...</li>
  </ul>
</details>
<details>
  <summary>6 packages built successfully:</summary>
  <ul>
    <li>python38Packages.beniget</li>
    <li>python38Packages.gast</li>
    <li>python38Packages.pythran</li>
    <li>python39Packages.beniget</li>
    <li>python39Packages.gast</li>
    <li>python39Packages.pythran</li>
  </ul>
</details>
<details>
  <summary>6 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md">maintainers-missing</a>

Package does not have a maintainer. Consider adding yourself?

Near pkgs/development/python-modules/beniget/default.nix:20:3:
```
   |
20 |   meta = {
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/duplicate-check-inputs.md">duplicate-check-inputs</a>

Dependencies listed in `propagatedBuildInputs` should not be repeated in `checkInputs`.

Detected duplicates:

- python3.9-numpy-1.20.3

Near pkgs/development/python-modules/pythran/default.nix:63:3:
```
   |
63 |   checkInputs = [
   |   ^
```

Near pkgs/development/python-modules/pythran/default.nix:44:3:
```
   |
44 |   propagatedBuildInputs = [
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/no-uri-literals.md">no-uri-literals</a>

URI literals are deprecated.
Near pkgs/development/python-modules/pythran/default.nix:77:16:
```
   |
77 |     homepage = https://github.com/serge-sans-paille/pythran;
   |                ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md">missing-patch-comment</a>

Consider adding a comment explaining the purpose of this patch on the line preceeding.
Near pkgs/development/python-modules/pythran/default.nix:34:5:
```
   |
34 |     (substituteAll {
   |     ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md">maintainers-missing</a>

Package does not have a maintainer. Consider adding yourself?

Near pkgs/development/python-modules/pythran/default.nix:75:3:
```
   |
75 |   meta = {
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/duplicate-check-inputs.md">duplicate-check-inputs</a>

Dependencies listed in `propagatedBuildInputs` should not be repeated in `checkInputs`.

Detected duplicates:

- python3.8-numpy-1.20.3

Near pkgs/development/python-modules/pythran/default.nix:63:3:
```
   |
63 |   checkInputs = [
   |   ^
```

Near pkgs/development/python-modules/pythran/default.nix:44:3:
```
   |
44 |   propagatedBuildInputs = [
   |   ^
```
</li>
  </ul>
</details>

**Pull request merged**